src/main/java/net/glowstone/util/Wearable.java

```diff
@@ -0,0 +1,120 @@
+package net.glowstone.util;
+
+import org.bukkit.enchantments.EnchantmentTarget;
+import org.bukkit.Material;
+import java.util.Map;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.inventory.ItemStack;
+import java.util.HashMap;
+import java.util.Set;
+
+public final class Wearable {
```

Comment:
After seeing this class, I have understood what you're trying to do. This behavior is already implemented in `GlowPlayerInventory`. Use `tryToFillSlots(armor, 36, 40)`

I have put some comments about your code in case you need to use similar Java code in the future.